test/alternator/util.py

```diff
@@ -206,3 +208,17 @@ def client_no_transform(client):
 
 def is_aws(dynamodb):
     return dynamodb.meta.client._endpoint.host.endswith('.amazonaws.com')
+
+# Tries to inject an error via Scylla REST API. It only works on Scylla,
+# and only in specific build modes (dev, debug, sanitize), so this function
+# is not guaranteed to have any effect.
+def maybe_inject_error(dynamodb, err, one_shot=False):
+    if dynamodb.meta.client._endpoint.host.endswith('.amazonaws.com'):
+        return
```

Comment:
I'm not thrilled about something not being tested with even knowing that it's not being tested. I prefer what we did in with_tracing() in test_tracing.py - where we call `pytest.skip()` the test if we can't enable that feature.  We can call the function `scylla_inject_error()` to emphasize it is Scylla-only.
However this will mean you'll need a second test (or to avoid code duplication, the same test with a pytest.mark.parameterize) that will  do the injection and that test will be skipped when we can't inject.

Reply:
I generally agree, but this case is somewhat specific. DynamoDB will also exercise the code path for UnprocessedKeys, but for a different reason - hitting its batch limits. We don't have those in Scylla, so that's why I slipped in an error injection point.